### ISSUE TITLE
Update a finding aid when one of its components is unpublished or suppressed

### ISSUE THREAD

**Issue by corylown** [sul-dlss/stanford-arclight#556]
See https://github.com/sul-dlss/stanford-arclight/issues/551

We need change how we're polling for changes to resource records. Under some conditions (using the "publish all" or "suppress" buttons in the ASpace UI) only the component's system modified time is changed. To detect this change we need to broaden our query changes to any resource, find all the resource ids, and they query for resources that are published, not suppressed, have an ead id, etc, to know that we should download a new copy of the complete finding aid.

**Comment by taylor-steve** [comment]
TLDR: This issue is fixed (mostly?) in ArchivesSpace v3.5.0.

@corylown Regarding the question of "why doesn't the TouchRecords mixin address this", I think this is what is going on:

For scoping tree ops (`/repositories/:repo_id/resources/:id/publish`, `/repositories/:repo_id/resources/:id/suppressed`) it seems like they are altering the database directly:
https://github.com/archivesspace/archivesspace/blob/fb093d6f2c69b71fe77ca8a67203cb8756e318b1/backend/app/model/ASModel_scoping.rb#L53

This doesn't trigger any of the hooks in touch records (whereas editing in the form would trigger `update_from_json`):
https://github.com/archivesspace/archivesspace/blob/fb093d6f2c69b71fe77ca8a67203cb8756e318b1/backend/app/model/mixins/touch_records.rb


Clearly, they know something of this, as v3.5.0 and higher have specific workarounds for publish. For example for AOs:

https://github.com/archivesspace/archivesspace/blob/fb093d6f2c69b71fe77ca8a67203cb8756e318b1/backend/app/model/archival_object.rb#L128
```ruby
  def publish!(setting = true)
    super(setting)
    Resource.update_mtime_for_ids([self.root_record_id])
  end
```
